libraries/GCS_MAVLink/examples/routing/routing.cpp

```diff
@@ -122,5 +123,10 @@ void loop(void)
     hal.scheduler->delay(1000);
 }
 
+/* dummy methods to avoid having to link against AP_Camera */
```

Comment:
When will they start making smart compilers? If get_camera is nullptr we'll never going to call these methods! 😄 

In a more serious note, this only worries me because it means that disabling features like CAMERA in vehicles won't result in any meaningful size reduction since the library will still be linked in. Disabling features is probably not used much (and I would even say to remove it), but maybe we should start thinking how we can make waf support enabling/disabling features.